map-storage/src/Upload/UploadController.ts

```diff
@@ -121,6 +121,21 @@ export class UploadController {
                 if (limiter.activeCount === 0 && limiter.pendingCount === 0) {
                     this.uploadLimiter.delete(virtualDirectory);
                 }
+
+                const cachePath = `cache/${virtualDirectory}`;
+                const cacheFileName = "cached-map-names.txt";
+                const files = await fileSystem.listFiles(virtualDirectory, ".json");
+
+                fs.mkdir(path.dirname(cachePath), { recursive: true }, (err) => {
+                    if (err) {
+                        throw new Error(err.message);
+                    }
+                });
+                fs.writeFile(`${cachePath}${cacheFileName}`, JSON.stringify(files), (err) => {
+                    if (err) {
+                        throw new Error(err.message);
+                    }
+                });
```

Comment:
This won't scale correctly.

You are writing the cache file on a local disk. So if we have 2 map-storage, the cache will only be available in one map-storage container.

You could however write the cache directly using the configured filesystem - so either DiskFileSystem (that does not scale) or S3FileSystem (that scales)

You would need to add a "writeFile" method to those file systems I guess.